### DISCUSSION TITLE
Error for corfu--auto-complete-deferred with new ispell capf emacs 30.0.5

### DISCUSSION THREAD

**Discussion by lukeflo** [minad/corfu#457] [Q&A] [answered]
I recently set up my laptop from scratch with void linux. I compiled emacs from source and installed my `init.el`.

So far everything works fine, also `corfu`. Only when I work in a `text-mode` derived mode I get the following error message while writing regular text:

```
Error running timer ‘corfu--auto-complete-deferred’: (error "ispell-lookup-words: No plain word-list found at systemdefault locations.  Customize ‘ispell-alternate-dictionary’ to set yours.")
```

It might have something to do with a change in commit [47e313e9805](https://github.com/emacs-mirror/emacs/commit/47e313e9805). From this point on, `ispell` seems to be invoked as regular `capf` function. The [fallback-solution](https://github.com/emacs-mirror/emacs/commit/47e313e9805#diff-731c852b7c27c6ee90c44c871e2988ecfc2d225bc720156d7a47c1d563c2683a) doesn't work. 

It seems related to this [issue](https://github.com/minad/corfu/issues/319) and this [one](https://github.com/company-mode/company-mode/issues/1437) using `company`. 

I also tried setting the different `ispell` dictionry vars, but it is a little bit confusing which is needed.

I would like to avoid to rebuild emacs or clear the whole `eln-cache`. This should only be the last solution.

My emacs properties:

GNU Emacs 30.0.50 (build 1, x86_64-pc-linux-gnu, X toolkit, cairo
 version 1.18.0, Xaw scroll bars) of 2024-05-02

I built with `--with-native-compilation=aot`, which might have an influence on the `eln-cache`.

Since I'm not an elisp pro, any help is appreciated.

**Comment by minad** [comment, marked as answer]
I suggest this configuration:

~~~ elisp
(setopt text-mode-ispell-word-completion nil)
~~~

**Reply by lukeflo** [reply]
Thank you for the fast reply. That did the trick. In the linked commit only `text-mode-meta-tab-ispell-complete-word` is mentioned which didn't exist. So thanks for the hint. Might be due to the development status of emacs 30.0.5 that some variables can be renamed without knowing...

**Reply by kflak** [reply]
This doesn't seem to work anymore on 31.0.50 unfortunately.

EDIT: seems to work now after deleting and reinstalling corfu. Also it seems that text-mode-ispell-word-completion needs to be set through the customize interface:

```emacs-lisp
(customize-set-variable 'text-mode-ispell-word-completion nil)
```

**Comment by yagossc** [comment]
In Arch Linux I was simply missing the "words" package:
```
extra/words 2.1-8
    A collection of International 'words' files for /usr/share/dict
```
